membrane_curvature/lib/mods.py

```diff
@@ -160,27 +54,24 @@ def core_fast_leaflet(z_Ref, leaflet, traj, jump, n_cells, lipid_types,
 
     grid_count = np.zeros([n_cells, n_cells])
 
-    for frame in range(0, traj.n_frames, jump):
+    for ts in universe.trajectory:
+
         grid_1 = np.zeros([n_cells, n_cells])
         grid_2 = np.zeros([n_cells, n_cells])
 
         factor = np.float32(n_cells / max_width)
 
-        for lipid_type in lipid_types:
-
-            for bead in lipid_ref[leaflet][lipid_type]:
-
-                x, y, z = traj.xyz[frame, bead, :]
+        for bead in selection:
+            x, y, z = bead.position/10
 
-                l = int(abs(x) * factor)
-                m = int(abs(y) * factor)
+            index_l, index_m = grid_map((x, y), factor)
 
-                try:
-                    grid_1[l, m] += z
-                    grid_2[l, m] += 1
+            try:
+                grid_1[index_l, index_m] += z
+                grid_2[index_l, index_m] += 1
 
-                except BaseException:
-                    pass
+            except ValueError:
+                print("Atom outside grid boundaries. Skipping atom {}".format(bead))
```

Comment:
We tend to avoid `print` in production code as it's easy overlooked and people often don't like it when their screen is filled with random messages. If it needs to be printed then we use a logger. If you need to know provide warnings to the user then you can also build up the information in a list and then outside the loop summarize what happened in a message.

For right now, you can leave the `print` but down the line you'll want to think about hoe to communicate with the user.